backend/routes/graph.py

```diff
@@ -93,56 +98,35 @@ async def concept_graph(
     verbose and timer('concept_graph()')
 
     # Get concepts & metadata
-    concepts_raw: List[Dict] = [dict(x) for x in get_cset_members_items(
-        codeset_ids=codeset_ids, columns=['concept_id', 'vocabulary_id', 'standard_concept'])]
-    # Filter
-    # TODO: see: filter_concepts()
-    concepts: List[Dict]
+    concepts_unfiltered: List[RowMapping] = get_cset_members_items(
+        codeset_ids=codeset_ids, columns=['concept_id', 'vocabulary_id', 'standard_concept'])
+
+    # Fill gaps
+    missing_in_betweens_ids: Set[int] = get_missing_in_between_nodes(
+        REL_GRAPH, set([c['concept_id'] for c in concepts_unfiltered]), False)
+    missing_in_betweens: List[RowMapping] = get_concepts(missing_in_betweens_ids)
+    concepts_unfiltered2: List[RowMapping] = concepts_unfiltered + missing_in_betweens
+    # Filter: by vocab & non-standard
+    concepts: List[Dict[str, Any]]
     hidden_by_voc: Dict[str, Set[int]]
     nonstandard_concepts_hidden: Set
     concepts, hidden_by_voc, nonstandard_concepts_hidden = filter_concepts(
-        concepts_raw, hide_vocabs, hide_nonstandard_concepts)
-    concept_ids: Set[int] = set([c['concept_id'] for c in concepts])
-    # concept_ids.update(extra_concept_ids)  # future
-
-    # TODO: Joe: make use of: hidden_dict, nonstandard_concepts_hidden?
-    #  - possibly just return to frontend for reporting
-
-    # TODO: call get missing in between nodes
-    #  - Set[int] correct?
-    missing_in_betweens_ids: Set[int] = get_missing_in_between_nodes(REL_GRAPH, concept_ids)
-    # TODO: needs to be the list of missing nodes with vocab and nonstandard id. probably is get_concepts()
-    missing_in_betweens = []
-
-    concepts_m: List[Dict]
-    hidden_by_voc_m: Dict[str, Set[int]]
-    nonstandard_concepts_hidden_m: Set
-    concepts_m, hidden_by_voc_m, nonstandard_concepts_hidden_m = filter_concepts(
-        missing_in_betweens, hide_vocabs, hide_nonstandard_concepts)
-
-    # TODO: merge/update the 3 variables each from the 2 returns of filter_concepts()
-    #  - is the below correct?
-    concept_ids.update(missing_in_betweens_ids)
-    hidden_by_voc_m.update(hidden_by_voc)
-    nonstandard_concepts_hidden = nonstandard_concepts_hidden.union(nonstandard_concepts_hidden_m)
+        concepts_unfiltered2, hide_vocabs, hide_nonstandard_concepts)
 
     # Get subgaph
-    sg = REL_GRAPH.subgraph(concept_ids)
-
+    sg: DiGraph = REL_GRAPH.subgraph(set([c['concept_id'] for c in concepts]))
 
+    # Return
     verbose and timer('done')
-    # TODO: change to return: edges, filled gaps (missing_in_betweens), NOT layout, hidden_dict &
-    #  nonstandard_concepts_hidden ((merged of the normal and the  missing in betweens calls
-    #  maybe return: preferred_concept_ids, and orphans
-    #  - update function return typedef
-    return sg, missing_in_betweens, hidden_by_voc, nonstandard_concepts_hidden
+    return sg, missing_in_betweens_ids, hidden_by_voc, nonstandard_concepts_hidden
```

Comment:
## Orphans & Preferred concept IDs
Didn't return:
- `orphans`: no parent/child
- `preferred_concept_ids`: things we're trying to link in the graph; non-orphan items

Because Siggie is moving these calculations to the frontend. I think this is because this is easy to calculate on the frontend, whereas the rest has to be done on the backend.